galley/pkg/config/analysis/local/source.go

```diff
@@ -118,8 +124,16 @@ func (s *precedenceSource) Start() {
 		return
 	}
 
+	// Init expected counts map
+	s.expectedCounts = make(map[collection.Name]int)
+	for _, i := range s.inputs {
+		for _, c := range i.cols {
+			s.expectedCounts[c]++
+		}
```

Comment:
So that changes allows for source-to-collection to not be 1-to-1? When would that be the case?

Reply:
Before, each source was assumed to have the same set of collections. Now a source and its expected collections are provided in a pair, so we know exactly how many FullSyncs to expect for each collection. 

This needed to change so I could use mesh config sources alongside "regular" k8s resource sources, and expect the right count of FullSyncs for everything.